aten/src/ATen/native/cuda/BinaryOpsKernel.cu

```diff
@@ -67,13 +67,20 @@ static void div_kernel_cuda(TensorIterator& iter) {
 
 template <typename scalar_t>
 void mul_kernel_impl(TensorIterator& iter) {
-  gpu_binary_kernel(iter, []GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
-    return a * b;
-  });
+  if (iter.dtype() == ScalarType::Bool) {
+    gpu_binary_kernel(iter, []GPU_LAMBDA(bool a, bool b) -> bool {
```

Comment:
also a comment here about why this is implemented this way would be great.  e.g. is it because `*` on bool isn't defined, is it because the compiler warns because it is trying to be helpful, etc.